tests/unit/ip_router_interface/v0/test_ip_router.py

```diff
@@ -0,0 +1,487 @@
+# Copyright 2023 Ubuntu
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+import json
+import logging
+import os
+import shutil
+import unittest
+from itertools import chain
+
+from ops.testing import Harness
+
+from tests.provider_charm.src.charm import (
+    IP_ROUTER_PROVIDER_RELATION_NAME,
+    SimpleIPRouteProviderCharm,
+)
+from tests.requirer_charm.src.charm import (
+    IP_ROUTER_REQUIRER_RELATION_NAME,
+    SimpleIPRouteRequirerCharm,
+)
+
+logger = logging.getLogger(__name__)
```

Comment:
Do we need to log anything in unit tests?

Reply:
Not that I'm aware of.